pyanaconda/modules/storage/installation.py

```diff
@@ -68,6 +77,56 @@ def _report_message(self, data):
         """
         self.report_progress(data.msg)
 
+    def _wait_for_entropy(self, data):
+        """Wait for entropy.
+
+        :param data: Blivet's callback data
+        :return: True if we are out of time, otherwise False
+        """
+        log.debug(data.msg)
+        required_entropy = data.min_entropy
+        total_time = self._entropy_timeout
+        current_time = 0
+
+        while True:
+            # Report the current status.
+            current_entropy = get_current_entropy()
+            current_percents = min(int(current_entropy / required_entropy * 100), 100)
+            remaining_time = max(total_time - current_time, 0)
+            self._report_entropy_message(current_percents, remaining_time)
+
+            sleep(5)
+            current_time += 5
+
+            # Enough entropy gathered.
+            if current_percents == 100:
+                return False
+
+            # Out of time.
+            if remaining_time == 0:
+                return True
+
+    def _report_entropy_message(self, percents, time):
+        """Report an entropy message.
+
+        :param percents: the percentage of gathered entropy
+        :param time: a number of seconds of remaining time
+        """
+        if percents == 100:
+            self.report_progress(_("Gathering entropy 100%"))
+            return
+
+        if time == 0:
+            self.report_progress(_("Gathering entropy (time ran out)"))
+            return
+
+        message = _("Gathering entropy {percents}% (remaining time {time})").format(
+            percents=percents,
+            time=timedelta(seconds=time)
+        )
```

Comment:
Please change this to `if-elif-else` with `self.report_progress` on one place with `message` variable.